extensions/cornerstone/src/panels/PanelSegmentation.tsx

```diff
@@ -0,0 +1,286 @@
+import React from 'react';
+import { SegmentationTable } from '@ohif/ui-next';
+import { callInputDialog, colorPickerDialog, createReportAsync } from '@ohif/extension-default';
+import { useActiveViewportSegmentationRepresentations } from '../hooks/useActiveViewportSegmentationRepresentations';
+
+export default function PanelSegmentation({
+  servicesManager,
+  commandsManager,
+  extensionManager,
+  configuration,
+  children,
+}: withAppTypes) {
+  const { segmentationService, viewportGridService, uiDialogService, customizationService } =
+    servicesManager.services;
+
+  const { segmentationsWithRepresentations, disabled } =
+    useActiveViewportSegmentationRepresentations({
+      servicesManager,
+    });
+
+  const handlers = {
+    onSegmentationAdd: async () => {
+      segmentationService.createLabelmapForViewport(viewportGridService.getActiveViewportId());
+    },
+
+    onSegmentationClick: (segmentationId: string) => {
+      segmentationService.setActiveSegmentation(
+        viewportGridService.getActiveViewportId(),
+        segmentationId
+      );
+    },
+
+    onSegmentAdd: segmentationId => {
+      segmentationService.addSegment(segmentationId);
+    },
+
+    onSegmentClick: (segmentationId, segmentIndex) => {
+      segmentationService.setActiveSegment(segmentationId, segmentIndex);
+      segmentationService.jumpToSegmentCenter(segmentationId, segmentIndex);
+    },
+
+    onSegmentEdit: (segmentationId, segmentIndex) => {
+      const segmentation = segmentationService.getSegmentation(segmentationId);
+
+      if (!segmentation) {
+        return;
+      }
+
+      const segment = segmentation.segments[segmentIndex];
+      const { label } = segment;
+
+      callInputDialog(uiDialogService, label, (label, actionId) => {
+        if (label === '') {
+          return;
+        }
+
+        segmentationService.setSegmentLabel(segmentationId, segmentIndex, label);
+      });
+    },
+
+    onSegmentationEdit: segmentationId => {
```

Comment:
If you are going to have two very similar names, you need a comment to explain that one is segmentation overall which contains segments, and the other is an individual segment.  It took me a moment to remember that.
Also, this one should also be in a commands module.